addons/sale_mrp/models/mrp_production.py

```diff
@@ -17,6 +17,15 @@ def _compute_sale_order_count(self):
         for production in self:
             production.sale_order_count = len(production.procurement_group_id.mrp_production_ids.move_dest_ids.group_id.sale_id)
 
+    @api.depends('bom_id', 'product_id', 'procurement_group_id.mrp_production_ids.move_dest_ids.group_id.sale_id.analytic_account_id')
+    def _compute_analytic_distribution(self):
```

Comment:
Hello,

Thank you for your fix. However, using a compute in this case doesn't seem to be a good idea. The manufacturing order should directly receive the analytic distribution in the values upon creation. So, you can simply do this:

```diff
diff --git a/addons/sale_mrp/models/stock_move.py b/addons/sale_mrp/models/stock_move.py
index bc0a1f1bce09..2483937d3e7b 100644
--- a/addons/sale_mrp/models/stock_move.py
+++ b/addons/sale_mrp/models/stock_move.py
@@ -9,7 +9,7 @@ class StockMove(models.Model):
 
     def _prepare_procurement_values(self):
         res = super()._prepare_procurement_values()
-        res['analytic_account_id'] = self.sale_line_id.order_id.analytic_account_id
+        res['analytic_distribution'] = {self.sale_line_id.order_id.analytic_account_id.id: 100.0}
         return res
 
```

Reply:
I used a compute to also cover the case when a project (that has an analytic account) is created from the SO